### PR TITLE
(Tests) Feature/add project remove with wait for completion

### LINKED ISSUE #741: Testing project creation phase fails due to "unitt-cd already exists"
**Describe the bug**

When running `ods-core/tests>make test` the 'Testing project creation' phase failed because of:

```
Testing project creation
=== RUN   TestCreateODSJenkinsThruShellScripts
--- FAIL: TestCreateODSJenkinsThruShellScripts (1.14s)
    create-cd-jenkins_test.go:23: Execution of `create-project.sh` failed:
        StdOut: PROJECT_ID=unitt
        Create projects unitt-cd, unitt-dev and unitt-test

        StdErr: Error from server (AlreadyExists): project.project.openshift.io "unitt-cd" already exists
=== RUN   TestCreateNakedOCPProjectsThruShellScripts
--- FAIL: TestCreateNakedOCPProjectsThruShellScripts (0.85s)
    create-projects_test.go:23: Execution of `create-project.sh` failed:
        StdOut: PROJECT_ID=unitt
        Create projects unitt-cd, unitt-dev and unitt-test

        StdErr: Error from server (AlreadyExists): project.project.openshift.io "unitt-cd" already exists
=== RUN   TestCreateProjectThruWebhookProxyJenkinsFile
Waiting for build. Current status: Running
Waiting for build. Current status: Running
Waiting for build. Current status: Running
Waiting for build. Current status: Running
Waiting for build. Current status: Running
Waiting for build. Current status: Failed
--- FAIL: TestCreateProjectThruWebhookProxyJenkinsFile (156.08s)
    jenkinsfile_test.go:152: Actual jenkins stages from run: ods-corejob-create-project-unitt-master-1 don't match -golden:
        '[
          {
            "stage": "Checkout ods-core",
            "status": "SUCCESS"
          },
          {
            "stage": "Checkout ods-configuration",
            "status": "SUCCESS"
          },
          {
            "stage": "Create OpenShift projects",
            "status": "SUCCESS"
          },
          {
            "stage": "Setup Jenkins in CD project",
            "status": "SUCCESS"
          }
        ]
        '
        -jenkins response:
        '[]
        '
FAIL
FAIL    github.com/opendevstack/ods-core/tests/create-projects  158.585s
FAIL
Makefile:4: recipe for target 'test' failed
make: *** [test] Error 1
```

I guess a colleague of mine has already run this on our cluster, because he mentioned that before and it was my first time running that phase. 

It would be nice to be able to run that more than once.

Also, I don't understand that part in the error message, thus I'm not sure if it is caused by the same issue or not:
```
--- FAIL: TestCreateProjectThruWebhookProxyJenkinsFile (156.08s)
    jenkinsfile_test.go:152: Actual jenkins stages from run: ods-corejob-create-project-unitt-master-1 don't match -golden:
...
```

fyi @michaelsauter @georgfedermann @clemensutschig

### PR DESCRIPTION
**Pull request by clemensutschig**
fixes #741

### CODE REVIEW

**Comment by clemensutschig** [comment]
@georgfedermann  - feel free to try ... this should also allow to run make tests multiple times

**Comment by renedupont** [comment]
While you are working on the documentation... could we also add a description of that test caching mechanism?

I encountered today:
```
PASS
ok      github.com/opendevstack/ods-core/tests/ods-verify       (cached)
```

Not 100% sure how that caching works but what I figured out is that it relies on an existing `response.txt` in `ods-core/tests/ods-verify` from previous test runs to get activated. 

The issue I had, and it might even be considered as a bug, is that I changed the `PROV_APP_PROVISION_CLEANUP_INCOMPLETE_PROJECTS_ENABLED` value in my local `ods-configuration/ods-core.env` but did not do `tailor apply`, which would normally let the `make verify` fail. But because this response.txt was existing it just passed with the above (cached) message.

**Comment by clemensutschig** [comment]
@renedupont  - these two topics are unrelated - (cached) - is a go test feature 👍 
if you want to change it - just add ` -count=1` into verify.sh and installation-test.sh

File: tests/verify.sh [hunk 1]
```diff
@@ -3,6 +3,10 @@
 set +e
 set -o pipefail
 
+source ../../ods-configuration/ods-core.env
```
Comment (renedupont) [inline tests/verify.sh:6]: I get on my bash on ubuntu (via windows WSL):

```
Verifying ODS installation
../../ods-configuration/ods-core.env: line 248: syntax error near unexpected token `('
../../ods-configuration/ods-core.env: line 248: `PROV_APP_MAIL_PASSWORD=ENC(...)'
```

in ENC(...) the only special chars are + and =

Any idea?

**Comment by clemensutschig** [comment]
Yak. We have not Seen issues there... ;'ö

**Comment by renedupont** [comment]
I merged from ods/master into my local ods-core branch and ran ´make test´ again but still receive:

```
=== RUN   TestCreateODSJenkinsThruShellScripts
--- FAIL: TestCreateODSJenkinsThruShellScripts (1.24s)
    create-cd-jenkins_test.go:23: Execution of `create-project.sh` failed:
        StdOut: PROJECT_ID=unitt
        Create projects unitt-cd, unitt-dev and unitt-test

        StdErr: Error from server (AlreadyExists): project.project.openshift.io "unitt-cd" already exists
=== RUN   TestCreateNakedOCPProjectsThruShellScripts
--- FAIL: TestCreateNakedOCPProjectsThruShellScripts (0.78s)
    create-projects_test.go:23: Execution of `create-project.sh` failed:
        StdOut: PROJECT_ID=unitt
        Create projects unitt-cd, unitt-dev and unitt-test

        StdErr: Error from server (AlreadyExists): project.project.openshift.io "unitt-cd" already exists
=== RUN   TestCreateProjectThruWebhookProxyJenkinsFile
Buildname from response: ods-corejob-create-project-unitt-master-1
Waiting for build. Current status: Running
Waiting for build. Current status: Failed
--- FAIL: TestCreateProjectThruWebhookProxyJenkinsFile (392.96s)
    jenkinsfile_test.go:153: Actual jenkins stages from run: ods-corejob-create-project-unitt-master-1 don't match -golden:
        '[
          {
            "stage": "Checkout ods-core",
            "status": "SUCCESS"
          },
          {
            "stage": "Checkout ods-configuration",
            "status": "SUCCESS"
          },
          {
            "stage": "Create OpenShift projects",
            "status": "SUCCESS"
          },
          {
            "stage": "Setup Jenkins in CD project",
            "status": "SUCCESS"
          }
        ]
        '
        -jenkins response:
        '[]
        '
FAIL
FAIL    github.com/opendevstack/ods-core/tests/create-projects  395.502s
FAIL
Makefile:4: recipe for target 'test' failed
make: *** [test] Error 1

```

❓

File: tests/verify.sh (hunk 1 above)
Comment (clemensutschig) [reply tests/verify.sh:6]: @michaelsauter  - maybe you have an idea? .. this only happens on @renedupont 's shell ..

**Comment by renedupont** [comment]
Looks slightly better now, but I still get:

```
=== RUN   TestCreateNakedOCPProjectsThruShellScripts
project unitt-test deleted - project.project.openshift.io "unitt-test" deleted

project unitt-dev deleted - project.project.openshift.io "unitt-dev" deleted

project unitt-cd deleted - project.project.openshift.io "unitt-cd" deleted

--- FAIL: TestCreateNakedOCPProjectsThruShellScripts (28.87s)
    create-projects_test.go:23: Execution of `create-project.sh` failed:
        StdOut: PROJECT_ID=unitt
        Create projects unitt-cd, unitt-dev and unitt-test

        StdErr: Error from server (AlreadyExists): project.project.openshift.io "unitt-cd" already exists
```

File: tests/verify.sh (hunk 1 above)
Comment (michaelsauter) [reply tests/verify.sh:6]: That issue is not shell related. But the password may contain chars which get interpreted on the shell. Enclosing in `""` helps - e.g. ` PROV_APP_MAIL_PASSWORD="....."`

**Comment by clemensutschig** [comment]
Ok then we should fix this in ods-core.env Sample

Michael Sauter <notifications@github.com> schrieb am Mo., 3. Aug. 2020,
13:55:

**Pull request merged**